naga/src/valid/mod.rs

```diff
@@ -616,6 +614,7 @@ impl Validator {
                 .into_boxed_slice(),
         };
 
+        let mut array_override_expr_set = HandleSet::new();
```

Comment:
There's something about this check that makes me uncomfortable. It seems like either the UniqueArena will ensure the expressions are unique automatically, in which case the check will never fail, or you've got array types with distinct base/stride but the same expression, which doesn't seem necessary to forbid.

Do we have a test case that exercises this check?

Reply:
You are right, this validation is not needed. When thinking about your comment about `UniqueArena::replace` (https://github.com/gfx-rs/wgpu/pull/6787#discussion_r1927937857) I thought it was needed.

I will remove it and add a comment instead.